content/en/building/guides/interoperability/openmrs.md

```diff
@@ -0,0 +1,373 @@
+
+```bash
+curl -X GET localhost:8090/openmrs/ws/fhir2/R4/Patient/?identifier=[identifier] -H "Authorization: Basic $(echo -n admin:Admin123 | base64)"
+```
+
+This sequence diagrams shows the entire flow including the OpenMRS Mediator and the intermediate FHIR Server.
+![](cht-outgoing-patients.png)
+
+### Sending forms CHT->OpenMRS
+
+Any data sent from CHT to OpenMRS needs to map to a [concept](https://wiki.openmrs.org/display/docs/Concept+Dictionary+Basics) in OpenMRS. Each concept has a code which will be used to identify the concept in the CHT Application, the FHIR Server, and OpenMRS.
+For any fields to send to OpenMRS, first find or create matching concepts in OpenMRS.
+Then, using the appropriate codes, configure an outbound push as described in the [CHT FHIR config documentation]({{< ref "building/guides/interoperability/cht-config#outbound-reports" >}}).
+
+In OpenMRS, all form submissions are represented as `Home Visit` encounter types, with a `Visit Note` encounter.
+Any fields in the outbound push config are converted to FHIR observations, which are linked to the `Visit Note`.
+
+After setting up the outbound push, test that it works in the test environment by submitting a report to the form in the CHT application.
+Log in to OpenHIM and view the transaction log. You should see: 
+1. A request from the CHT application to the CHT Mediator, containing all the fields from the form that were mapped to concepts.
+    ![](cht-post-encounter.png)
+    ```json
+    {
+      "id": "442b0937-a32f-443e-8d28-7d9a7552fda2",
+      "patient_uuid": "75905106a4bc2a9046ed28df070016ce",
+      "reported_date": 1730355950419,
+      "observations": [
+        {
+          "code": "5090AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA",
+          "valueQuantity": {
+            "value": 175,
+            "unit": "cm"
+          },
+          "label": "Height in cm"
+        },
+        {
+          "code": "5089AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA",
+          "valueQuantity": {
+            "value": 60,
+            "unit": "kg",
+          },
+          "label": "Weight in kg"
+        }
+      ]
+    }
+    ```
+1. A PUT request to the FHIR server to create an encounter, and one request for each observation contained in that encounter.
+    ![](openhim-put-encounter.png)
+1. After a maximum of 1 minute, a polling request from the OpenMRS mediator, and related GET requests to the FHIR server and OpenMRS. The results from the FHIR server should include an encounter to represent the form submission, and observations for any fields that were mapped to concepts.
+    ![](openmrs-sync-encounter.png)
+1. Requests to OpenMRS to create a `Home Visit` encounter, a `Visit Note` encounter, and one request for each observation.
+    ![](openmrs-post-observation.png)
+1. A request to the FHIR server updating the encounter with the corresponding id from OpenMRS.
```

Comment:
Ok, after creating the CHT report, I see only two POST request and then all GET. 
This is how the transaction logs looks like:
![image](https://github.com/user-attachments/assets/24ce5c6a-4075-43be-8e57-188c3499f7c8)

Reply:
Resolving this conversation as I created [this](Failure to Sync Encounters and Observations from CHT to OpenMRS via Mediator](https://github.com/medic/cht-interoperability/issues/157#top) issue to follow up